### PR TITLE
Remove unused ABI definitions

### PR DESCRIPTION
**Pull request by vmoroz**
In this PR we remove several IDL files that are not used and may cause unnecessary confusion:
- IBackgroundEventListener.idl
- ILifecycleEventListener.idl
- NoExceptionAttribute.idl
- ReactInstance.idl

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/microsoft/react-native-windows/pull/5030)

### CODE REVIEW

**Comment by aeulitz** [comment]
>     <ClInclude Include="LifecycleState.h" />

Still needed?

---
Refers to: vnext/Microsoft.ReactNative/Microsoft.ReactNative.vcxproj:321 in e045c6c. [](commit_id = e045c6c75983b2492bcb995805769d99b646af2d, deletion_comment = False)

File: vnext/Microsoft.ReactNative/ReactInstance.idl [hunk 1]
```diff
@@ -1,18 +0,0 @@
...
-namespace Microsoft.ReactNative {
-
-  [webhosthidden]
-  interface IReactInstance {
-    void InvokeFunction(String moduleName, String method, IVectorView<IInspectable> arguments);
-  }
-
-  [webhosthidden]
-  [default_interface]
-  runtimeclass ReactInstance : IReactInstance {
```
Comment (aeulitz) [inline vnext/Microsoft.ReactNative/ReactInstance.idl:15]: >ReactInstance [](start = 15, length = 13)

For my education: What takes its place?

File: vnext/Microsoft.ReactNative/ReactNativeHost.h [hunk 2]
```diff
@@ -35,7 +34,6 @@ struct ReactNativeHost : ReactNativeHostT<ReactNativeHost> {
   Mso::CntPtr<Mso::React::IReactHost> m_reactHost;
 
   ReactNative::ReactInstanceSettings m_instanceSettings{nullptr};
```
Comment (aeulitz) [inline vnext/Microsoft.ReactNative/ReactNativeHost.h:36]: >m_instanceSettings [](start = 37, length = 18)

For my education: What do these settings now apply to?

File: vnext/Microsoft.ReactNative/ReactInstance.idl (hunk 1 above)
Comment (vmoroz) [reply vnext/Microsoft.ReactNative/ReactInstance.idl:15]: We do not expose the ReactInstance. Instead we offer the IReactContext.

---
In reply to: [431494732](https://github.com/microsoft/react-native-windows/pull/5030#discussion_r431494732) [](ancestors = 431494732)

File: vnext/Microsoft.ReactNative/ReactNativeHost.h (hunk 2 above)
Comment (vmoroz) [reply vnext/Microsoft.ReactNative/ReactNativeHost.h:36]: They apply for the ReactInstance used internally. We just do not expose it through the ABI.

---
In reply to: [431495017](https://github.com/microsoft/react-native-windows/pull/5030#discussion_r431495017) [](ancestors = 431495017)

**Comment by vmoroz** [comment]
>     <ClInclude Include="LifecycleState.h" />

Good point! Removed.

---
In reply to: [634992170](https://github.com/microsoft/react-native-windows/pull/5030#issuecomment-634992170) [](ancestors = 634992170)

---
Refers to: vnext/Microsoft.ReactNative/Microsoft.ReactNative.vcxproj:321 in e045c6c. [](commit_id = e045c6c75983b2492bcb995805769d99b646af2d, deletion_comment = False)

**Pull request merged**